unblob/pool.py

```diff
@@ -1,11 +1,29 @@
 import abc
 import multiprocessing as mp
+import signal
 import sys
 from multiprocessing.queues import JoinableQueue
 from typing import Any, Callable, Union
 
+from structlog import get_logger
+
 from .logging import multiprocessing_breakpoint
 
+logger = get_logger()
+
+
+class ShutDownRequired(Exception):
+    pass
+
+
+def _handle_signal(signum: int, _frame):
+    logger.debug(f"Signal ({signal.Signals(signum).name}) received")
```

Comment:
By default, Python will raise `KeyboardInterrupt` on CTRL-C. What is the added value here raising our own exception?

Reply:
Right, at this point, nothing, really. 
However we decided that we should handle SIGTERM as well so at this point, we will need some custom exception. And since we have that, we should used the same for SIGTERM and SIGINT.